src/CoreWCF.Http/src/CoreWCF/WS2007FederationHttpBinding.cs

```diff
@@ -0,0 +1,121 @@
+        //    : this()
+        //{
+        //    ApplyConfiguration(configName);
+        //}
+
+        public WS2007FederationHttpBinding()
+            : base()
+        {
+          //  this.ReliableSessionBindingElement.ReliableMessagingVersion = WS2007ReliableMessagingVersion;
+           // this.TransactionFlowBindingElement.TransactionProtocol = WS2007TransactionProtocol;
+            this.HttpsTransport.MessageSecurityVersion = WS2007MessageSecurityVersion;
+        }
+
+        public WS2007FederationHttpBinding(WSFederationHttpSecurityMode securityMode)
+            : this(securityMode, false)
+        {
+        }
+
+        public WS2007FederationHttpBinding(WSFederationHttpSecurityMode securityMode, bool reliableSessionEnabled)
+            : base(securityMode, reliableSessionEnabled)
+        {
+           // this.ReliableSessionBindingElement.ReliableMessagingVersion = WS2007ReliableMessagingVersion;
+           // this.TransactionFlowBindingElement.TransactionProtocol = WS2007TransactionProtocol;
+            this.HttpsTransport.MessageSecurityVersion = WS2007MessageSecurityVersion;
+        }
+
+        WS2007FederationHttpBinding(WSFederationHttpSecurity security, PrivacyNoticeBindingElement privacy, bool reliableSessionEnabled)
+            : base(security, privacy, reliableSessionEnabled)
+        {
+          //  this.ReliableSessionBindingElement.ReliableMessagingVersion = WS2007ReliableMessagingVersion;
+           // this.TransactionFlowBindingElement.TransactionProtocol = WS2007TransactionProtocol;
+            this.HttpsTransport.MessageSecurityVersion = WS2007MessageSecurityVersion;
+        }
+
+        //void ApplyConfiguration(string configurationName)
+        //{
+        //    WS2007FederationHttpBindingCollectionElement section = WS2007FederationHttpBindingCollectionElement.GetBindingCollectionElement();
+        //    WS2007FederationHttpBindingElement element = section.Bindings[configurationName];
+        //    if (element == null)
+        //    {
+        //        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ConfigurationErrorsException(
+        //            SR.GetString(SR.ConfigInvalidBindingConfigurationName,
+        //                         configurationName,
+        //                         ConfigurationStrings.WS2007FederationHttpBindingCollectionElementName)));
+        //    }
+        //    else
+        //    {
+        //        element.ApplyConfiguration(this);
+        //    }
+        //}
+
+        protected override SecurityBindingElement CreateMessageSecurity()
+        {
+            return this.Security.CreateMessageSecurity(false, WS2007MessageSecurityVersion);
+        }
+
+        //internal new static bool TryCreate(SecurityBindingElement sbe, TransportBindingElement transport, PrivacyNoticeBindingElement privacy, ReliableSessionBindingElement rsbe, TransactionFlowBindingElement tfbe, out Binding binding)
+        //{
```

Comment:
This and the method below are used when converting a wsdl back into a binding on the client side. Remove